dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/ArtifactDownloader.java

```diff
@@ -53,7 +59,32 @@
     private final ProgressBar progressBar = ProgressBar.getInstance();
     private static Map<String, Object> envMap = null;
 
-    public void synthesizeAndDownloadFeatures(List<String> mavenCoords, String dLocation, MavenRepository repository) throws InstallException {
+    final ExecutorService executor = Executors.newCachedThreadPool();
+
+    public Future<?> esaDownload(String coords, String dLocation, MavenRepository repository) {
```

Comment:
I like the idea of having `esaDownload` and `pomDownload` as separate functions but we could do better with code reuse: 

```
public Future<?> submitDownloadRequest(String coords, String fileType, String dLocation, MavenRepository repository)
{
    return executor.submit(() -> {
            try {
                synthesizeAndDownload(coords, fileType, dLocation, repository, false);
            } catch (InstallException e) {
                e.printStackTrace(); // throw this exception or log it 
            }
            return coords + fileType;
        });
}

public Future<?> esaDownload(String coords, String dLocation, MavenRepository repository) { 
   return submitDownloadRequest(coords, "esa", dLocation, repository, false); 
}

public Future<?> pomDownload(String coords, String dLocation, MavenRepository repository) { 
   return submitDownloadRequest(coords, "pom", dLocation, repository, false); 
}
```